userland/test-memory/main.c

```diff
@@ -5,41 +5,55 @@ inline unsigned char* getpage(unsigned int i) {
   return (unsigned char*) ((1 << i) * 4096L);
 }
 
-inline unsigned char getent(unsigned int i, unsigned int j) {
-  return ((i * 37) ^ j) & 0xff;
+inline unsigned char getent(unsigned int i, unsigned int j, int pid) {
+  return ((i * 37) ^ j ^ pid) & 0xff;
 }
 
 void main() {
-  // TODO: Use our PID or a rand() to seed the values stored in our allocated
-  // pages, and run two of these at the same time (after clone() or fork()).
-  DEBUG("Writing");
+  DEBUG("Allocating");
   const unsigned int num_pages = 7;
   for (unsigned int i = 0; i < num_pages; ++i) {
-    // Use unsigned char to prevent problems when extending to unsigned longs.
     unsigned char* my_page = getpage(i);
     int ret = palloc(my_page);
     if (ret) {
       DEBUG("Failed to allocate page!");
       exit();
     }
+  }
+
+  DEBUG("Forking");
```

Comment:
This program could more robustly test the fork syscall by testing the cases of:

- Making sure the kernel allows more than one fork, ever.
- Making sure the kernel allows both the parent and child to fork correctly after forking once.
- Making sure the kernel allows more than one child fork (bottom half) to be pending at once.